### ISSUE TITLE
Initialize weights of a module

### ISSUE THREAD

**Issue by Gurnek** [patrick-kidger/equinox#94]
Hello,
is there a way to initialize the weights of a module. For instance I have the code

`
class Update(eqx.Module):
    l1: eqx.Module
    l2: eqx.Module

    def __init__(self, key):
        k1, k2 = jrandom.split(key, 2)

        self.l1 = eqx.nn.Conv2d(48, 128, kernel_size=3, padding=1, use_bias=False, key=k1)
        self.l2 = eqx.nn.Conv2d(128, 16, kernel_size=3, padding=1, use_bias=False, key=k2)
        self.l2.weight = self.l2.weight.at[:].set(0.0)

    def __call__(self, state_grid):
        state_grid = jax.nn.relu(self.l1(state_grid))
        return self.l2(state_grid)
`
but this throws the FrozenInstanceError for modifying the weight value.
Is there a way to do this?

**Comment by patrick-kidger** [comment]
Yep: use `eqx.tree_at` to mutate `l2` out-of-place:

```python
self.l2 = eqx.tree_at(lamda l: l.weight, self.l2, replace_fn=jnp.zeros_like)
```

**Comment by Gurnek** [comment]
Thank you!
